src/Trinity.Core/Storage/Storage.cs

```diff
@@ -194,11 +194,11 @@ static Storage()
         #endregion
 
         #region Message Sending Interfaces
-        internal abstract void SendMessage(TrinityMessage message);
-        internal abstract void SendMessage(TrinityMessage message, out TrinityResponse response);
+        protected internal abstract void SendMessage(TrinityMessage message);

```

Comment:
Message passing interfaces may require further polishing:
1. The ability to send a message to a cell (chunk routing)
2. The ability to set message passing semantics (RR/All/LB/FirstAvailable etc.)
3. The ability to create a "transaction": a transaction is a set of servers, that every message to one partition(AG) will be deterministically sent to a server in the transaction set. This guarantees that the messages within a transaction effectively share a context, without excessive synchronization. When the transaction ends, we can further "commit" this to the whole memory cloud.

Reply:
@yatao All of the things you enumerate here are of critical importance particularly in our use-case and application area of Ontological Semantic Reasoning in what we call Multi-Zoned Ontology  (multiple AGs) with Global Semantic Event projection and Query/Answer Reasoning. 

In an actual product scenario like ours,  client's *KBs in the SF/GE Distributed (RDF+) member cloud is logically partition via secured licensed ID but the low-level distributed parsing, reasoning is local to entire cluster(s). So I need to be able to send(message) data using a variety of methods (i.e., direct message, Azure Events, Reliable Queues...).

*KB - def. W3C OWL 2 Full/DL Representation of Knowledge
*(RDF+) - def. Our Extended RDF Theoretic Graph Representation and Grafting Engine with FanoutSearch and extended/enhanced LIKQ capabilities.